signal_scanner_bot/transport.py

```diff
@@ -31,36 +32,76 @@ def on_status(self, status: tweepy.Status):
 
 
 ################################################################################
-# Twitter-to-signal
+# Twitter-to-Queue
 ################################################################################
-def _twitter_to_signal():
+def _twitter_to_queue():
     """
     Top level function for running the twitter-to-signal loop.
     """
     api = twitter.get_api()
     stream = tweepy.Stream(auth=api.auth, listener=Listener())
-    log.info("Stream initialized, starting to follow")
+    log.info("Twitter stream initialized, starting to follow")
     stream.filter(track=env.RECEIVE_HASHTAGS, stall_warnings=True)
 
 
-async def twitter_to_signal():
+async def twitter_to_queue():
     """
     Asynchronous interface for the twitter-to-signal loop.
     """
     loop = asyncio.get_event_loop()
     with concurrent.futures.ThreadPoolExecutor() as pool:
         try:
-            await loop.run_in_executor(pool, _twitter_to_signal)
+            await loop.run_in_executor(pool, _twitter_to_queue)
+            await asyncio.sleep(5)
         except Exception as err:
-            log.error("Exception occurred, halting process")
+            log.error(
+                "Exception occurred in the Twitter to Queue pipeline, halting process"
+            )
             log.exception(err)
             signal.panic(err)
             env.STATE.STOP_REQUESTED = True
             raise
 
 
 ################################################################################
-# Signal-to-twitter
+# Queue-to-Signal
+################################################################################
+def _queue_to_signal():
```

Comment:
I think we may actually be able to make this function asyncio directly without having to do the whole "execute-in-thread-loop" that we're having to do for the twitter stream tracking by using an `asyncio.Queue`: https://docs.python.org/3/library/asyncio-queue.html

Reply:
This will change how items are added to the queue in the twitter-to-queue loop though, since that function is technically synchronous...hopefully that doesn't become too dizzying! It sucks that tweepy's interface isn't truly async compatible :/

Reply:
Honestly now might be the time to move to a twitter API package that supports asyncio more explicitly, like [peony](https://github.com/odrling/peony-twitter)

Reply:
Oh, I didn't even think about a different Twitter API library, that might be a good thing to investigate. I'll look into it.

As for the asyncio Queue class, I'll give that a shot and see how it works. I'm a little unclear on how exactly that will work but I imagine I'll be able to figure it out with some digging.